core/trino-spi/src/main/java/io/trino/spi/block/EncoderUtil.java

```diff
@@ -277,4 +278,81 @@ static void compactLongsWithNullsScalar(SliceOutput sliceOutput, long[] values,
         sliceOutput.writeInt(compactedIndex);
         sliceOutput.writeLongs(compacted, 0, compactedIndex);
     }
+
+    static void writeOffsetsWithNullsCompacted(SliceOutput sliceOutput, int[] rawOffsets, @Nullable boolean[] valueIsNull, int baseOffset, int positionCount)
+    {
+        // inlined from Objects.checkFromIndexSize (this is necessary for performance)
```

Comment:
I don't follow why. We've been using `checkFromIndexSize` in lots of performance sensitive code paths.
This isn't even inside a hit loop, so I don't get what is the benefit of manual inlining for this.
Do JMH benchmarks actually show a significant difference from manually inlining this ?